### PR TITLE
Refactor messaging sample and readme

### PR DESCRIPTION
**Pull request by wujack778**
Refactor messaging sample and readme.
[task link](https://dev.azure.com/azure-spring-integration/spring-integration-private/_sprints/taskboard/spring-integration-private%20Team/spring-integration-private/185?workitem=1079)
[issue 19848](https://github.com/Azure/azure-sdk-for-java/issues/19848)

### CODE REVIEW

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging/README.md [hunk 1]
```diff
@@ -1,15 +1,64 @@
-# Azure Spring Cloud Messaging Sample client library for Java
+# Spring Cloud Azure Messaging Sample shared library for Java
 
 ## Key concepts
+
+This code sample demonstrates how to use [AzureMessageListener.java][annotation-azure-message-listener] to listen to events from event hub.
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging/README.md:5]: “Event Hubs” I think

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging/src/main/java/com/azure/spring/sample/messaging/WebController.java [hunk 2]
```diff
@@ -17,8 +17,8 @@
 @RestController
 public class WebController {
 
-    private static final String EVENT_HUB_NAME = "eventhub1";
-    private static final String CONSUMER_GROUP = "cg1";
+    private static final String EVENT_HUB_NAME = "event-0329-hub";
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging/src/main/java/com/azure/spring/sample/messaging/WebController.java:20]: will "event-0329-hub" be created automaticly or need create before configuration

Comment (backwind1233) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging/src/main/java/com/azure/spring/sample/messaging/WebController.java:20]: do we need to write something about this in readme.md?

Comment (wujack778) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-messaging/src/main/java/com/azure/spring/sample/messaging/WebController.java:20]: Yes i agree

**Review by saragluna — APPROVED** [review]
LGTM.

**Comment by saragluna** [comment]
The tests kept failing because of azure-messaging-servicebus, I'll override the results since this PR only changed a sample.

**Pull request merged**